src/append_only_data.rs

```diff
@@ -663,6 +663,28 @@ where
     P: Permissions + std::hash::Hash + Clone,
 {
     fn append(&mut self, entries: &[(Vec<u8>, Vec<u8>)], last_entries_index: u64) -> Result<()> {
+        let existing: BTreeSet<Vec<u8>> = self
+            .inner
+            .data
+            .iter()
+            .map(|tuple| tuple.0.clone())
```

Comment:
I think we should be able to avoid cloning the keys of the entries if we make the type of `existing` and `new` `BTreeSet<&Vec<u8>>`.

Also, a minor point: to make the code potentially more readable, you could change the captured variable from `tuple` to `(key, _value)`, then rather than calling `tuple.0` you'd just have `key`.  Not a big deal for something as simple as this scenario though.